ledger/complete/wal/wal.go

```diff
@@ -204,23 +215,36 @@ func (w *DiskWAL) replay(
 		}
 	}
 
-	if loadedCheckpoint == -1 && startSegment == 0 {
-		hasRootCheckpoint, err := checkpointer.HasRootCheckpoint()
-		if err != nil {
-			return fmt.Errorf("cannot check root checkpoint existence: %w", err)
-		}
-		if hasRootCheckpoint {
-			flattenedForest, err := checkpointer.LoadRootCheckpoint()
+	if loadedCheckpoint == -1 {
+		if startSegment == 0 {
+			hasRootCheckpoint, err := checkpointer.HasRootCheckpoint()
 			if err != nil {
-				return fmt.Errorf("cannot load root checkpoint: %w", err)
+				return fmt.Errorf("cannot check root checkpoint existence: %w", err)
 			}
-			err = checkpointFn(flattenedForest)
-			if err != nil {
-				return fmt.Errorf("error while handling root checkpoint: %w", err)
+			if hasRootCheckpoint {
+				w.log.Info().Msgf("loading root checkpoint")
+
+				flattenedForest, err := checkpointer.LoadRootCheckpoint()
+				if err != nil {
+					return fmt.Errorf("cannot load root checkpoint: %w", err)
+				}
+				err = checkpointFn(flattenedForest)
+				if err != nil {
+					return fmt.Errorf("error while handling root checkpoint: %w", err)
+				}
+
+				w.log.Info().Msgf("root checkpoint loaded")
+				checkpointLoaded = true
 			}
 		}
 	}
 
+	// if should load checkpoint, but not loaded, then return error
+	if useCheckpoints && !checkpointLoaded {
+		return fmt.Errorf("no checkpoint file was found, tried all checkpoint files, but none of them is valid: %v",
+			allCheckpoints)
+	}
```

Comment:
this is to fix confusing error message like this:
```
trie with the given rootHash d21b0c1f3653cf074a35f365e70c1840ce5a017287715fca60e1edd5758c20e8 not found
```

into this:
```
could not replay segments [0:11]: no checkpoint file was found, try all checkpoint files, but they are invalid: []
```

Reply:
It seems we allow starting EN without checkpoint? So the checks here might be wrong. I'm going to revert this change